### PR TITLE
fix(ingest): fix missing snowflake lineage when table_pattern is set

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py [hunk 1]
```diff
@@ -37,6 +37,9 @@ def get_dataset_identifier(
     ) -> str:
         ...
 
+    def get_dataset_identifier_from_qualified_name(self, qualified_name: str) -> str:
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py:40]: Does this have any implementation or why it is just ...?

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py:40]: Hey, this Protocol class is only for mypy. This is required if using mixin classes, without actually doing inheritance. Reference  - https://mypy.readthedocs.io/en/latest/more_types.html#mixin-classes

Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py:40]: Can you add this comment in the code as well?

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py:40]: yes

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py:42]: done

**Review by treff7es — APPROVED** [review]
lgtm

**Pull request merged**